jetbrains-core/tst/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileAssumeRoleProviderTest.kt

```diff
@@ -0,0 +1,192 @@
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, role)
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.roleArn()).isEqualTo(role)
+        }
+    }
+
+    @Test
+    fun `duration_seconds gets respected if provided`() {
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, aString())
+            put(ProfileProperty.DURATION_SECONDS, "12345")
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.durationSeconds()).isEqualTo(12345)
+        }
+    }
+
+    @Test
+    fun `duration_seconds uses default if not provided`() {
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, aString())
+            put(ProfileProperty.DURATION_SECONDS, "abc")
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.durationSeconds()).isEqualTo(3600)
+        }
+    }
+
+    @Test
+    fun `duration_seconds uses default if invalid format`() {
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, aString())
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.durationSeconds()).isEqualTo(3600)
+        }
+    }
+
+    @Test
+    fun `MFA is prompted if keys are specified`() {
+        val mfaSerial = aString()
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, aString())
+            put(ProfileProperty.MFA_SERIAL, mfaSerial)
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.tokenCode()).isEqualTo(mfaToken)
+        }
+    }
+
+    @Test
+    fun `external ID is respected if provided`() {
+        val id = aString()
+        val profile = profile {
+            put(ProfileProperty.ROLE_ARN, aString())
+            put(ProfileProperty.EXTERNAL_ID, id)
+        }
+
+        ProfileAssumeRoleProvider(stsClient, parentProvider, profile).resolveCredentials()
+
+        argumentCaptor<AssumeRoleRequest> {
+            verify(stsClient).assumeRole(capture())
+
+            assertThat(firstValue.externalId()).isEqualTo(id)
+        }
+    }
+
+    @Test
+    fun `resole session name is respected if provided`() {
```

Comment:
nit: test name should be `role session name`?

Reply:
yep